File: C++/Reverse Linked List.cpp
class Solution
{
public:
    ListNode *reverseList(ListNode *head)
    {
        ListNode *prev = nullptr;
        ListNode *curr = head;
        ListNode *nextNode = nullptr;

        while (curr != nullptr)
        {
            nextNode = curr->next;
            curr->next = prev;
            prev = curr;
            curr = nextNode;
        }
        return prev;
    }
};

File: 206 Reverse Linked List/Reverse Linked List.js
var reverseList = function (head) {
  let prev = null;
  let curr = head;
  let nextNode = null;

  while (curr !== null) {
    nextNode = curr.next; //2     3
    curr.next = prev; //2 ->  1-> null
    prev = curr; //4-> 3->2-> 1-> null
    curr = nextNode; //2 //3
  }
  return prev;
};
